client/signup/steps/site-picker/index.jsx

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import { getSite } from 'state/sites/selectors';
+import SiteSelector from 'components/site-selector';
+import StepWrapper from 'signup/step-wrapper';
+import SignupActions from 'lib/signup/actions';
+
+class SitePicker extends Component {
+	handleSiteSelect = ( siteSlug ) => {
+		const {
+				stepSectionName,
+				stepName,
+				goToStep,
+			} = this.props,
+			site = this.props.getSelectedSite( siteSlug ),
+			hasPlan = site && site.plan && site.plan.product_slug !== 'free_plan';
+
+		SignupActions.submitSignupStep(
+			{
+				stepName,
+				stepSectionName,
+				siteId: site.ID,
+				siteSlug: site.slug
+			},
+			[],
+			{}
+		);
+
+		SignupActions.submitSignupStep( { stepName: 'themes', wasSkipped: true }, [], {
+			themeSlugWithRepo: 'pub/twentysixteen'
+		} );
+
+		if ( hasPlan ) {
+			SignupActions.submitSignupStep( { stepName: 'plans', wasSkipped: true }, [], { cartItem: null, privacyItem: null } );
+
+			goToStep( 'user' );
+		} else {
+			goToStep( 'plans' );
+		}
+	};
+
+	filterSites = ( site ) => {
+		return site.capabilities.manage_options && ! site.jetpack;
+	};
+
+	renderScreen() {
+		return (
+			<Card className="site-picker__wrapper">
+				<SiteSelector
+					filter={ this.filterSites }
+					onSiteSelect={ this.handleSiteSelect }
+				/>
+			</Card>
+		);
+	}
+
+	render() {
+		return (
+			<StepWrapper
+				flowName={ this.props.flowName }
+				stepName={ this.props.stepName }
+				positionInFlow={ this.props.positionInFlow }
+				fallbackHeaderText={ this.props.headerText }
+				fallbackSubHeaderText={ this.props.subHeaderText }
+				signupProgress={ this.props.signupProgress }
+				stepContent={ this.renderScreen() } />
+		);
+	}
+}
+
+export default connect(
+	( state ) => {
+		return {
+			getSelectedSite: ( siteId ) => getSite( state, siteId )
```

Comment:
The idea here is that any time the store updates, mapStateToProps is called. We can use this function to create properties derived from global and prop state that will be merged with the components current prop state. It's ideal to avoid passing through functions for the reasons @mcsf listed in #14024. See also https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options
```jsx
const mapStateToProps = function( state ) {
    const siteId = getSelectedSiteId( state );
    return {
        siteId,
        siteSlug: getSiteSlug( state, siteId),
        hasPlan: isSiteOnPaidPlan( state, siteId ),
    };
}
connect( mapStateToProps )( SitePicker );
```

In the component you can then call any props you passed through using mapStateToProps.

For example
```jsx
render() {
    const { siteId, siteSlug, hasPlan } = this.props;
}
```